### DISCUSSION TITLE
Is it possible to unbind a keybinding?

### DISCUSSION THREAD

**Discussion by imochoa** [i3/i3#5013] [Q&A] [answered]
## Context
I just switched to using Regolith2 and their default i3 config (at `~/.config/regolith2/i3/config`) allows users to extend it by adding their files under a directory `~/.config/regolith2/i3/config.d/` 
Those extra files are included by a few lines at the end of the default i3 config file:
```
# Include any user i3 partials
include $HOME/.config/regolith2/i3/config.d/*
```
That's a nice way to separate the defaults from the user's customization. 

## Problem
Unless I'm mistaken, there's no way to overwrite an existing keybinding. So the approach the Regolith team took only lets you add new bindings and not modify existing ones.

For example, if they bind a key to switch layouts and I want to switch `tabbed->stacked`:
```
bindsym $mod+$i3-wm.binding.layout_mode layout toggle tabbed splith splitv
#                                                     ^^^^^^^
```
If I try to overwrite that later in an included file with:
```
bindsym $mod+$i3-wm.binding.layout_mode layout toggle stacked splith splitv
#                                                     ^^^^^^^
```

i3 complains, that they keybinding is already used. Which is fair, because it is 😂

## Workaround
I've been looking in the docs for a way to "unbind" a keybinding, so that I can overwrite it in my extra i3 config without messing with the regolith defaults. But it doesn't seem to be supported, right?

Or is there perhaps another way of doing it?

If not, it's not a big issue. I'll just modify `~/.config/regolith2/i3/config`. But I haven't found any discussion threads on it either, and I thought it might be an interesting language feature to add

**Comment by Airblader** [comment, marked as answer]
This unfortunately isn't possible currently. It's also a problem not restricted to bindings, window rules also cannot be undefined. I don't think it's worth doing anything from the i3 side here due to the complexity this would bring in for us.

**Reply by imochoa** [reply]
Ok I see. Thanks for the quick reply :)
